docs/adr/002-monolith-architecture-with-modular-subsystems.md

```diff
@@ -0,0 +1,75 @@
+# Monolith Architecture with Modular Subsystems
+
+## Context and Problem Statement
+
+The Nuts Knooppunt needs to integrate multiple subsystems including authentication (Nuts node), authorization (OPA), addressing (mCSD), localization, and consent management. Currently, these would require deploying and managing multiple containers, which adds operational complexity for parties deploying a Knooppunt. We need to decide on the overall architecture approach to favor simplicity and "plug and play" deployment while maintaining flexibility for future scaling.
```

Comment:
I'd like to add that it's also to support flexibility in target environments, regarding existing infrastructure. Some parties already host a Nuts node, or might want to host it separately. This could apply to the other subsystems as well.